application: serial_lte_modem Handle TCP/UDP datamode requirements (#3616)
nrfconnect/sdk-nrf: applications/serial_lte_modem/doc/slm_description.rst

   This option specifies the poll time-out for the TCP connection, in seconds.

.. option:: CONFIG_SLM_TCP_CONN_TIME - Connection time-out in seconds for TCP server

   This option specifies the connection time-out for the TCP connection, in seconds.

.. option:: CONFIG_SLM_DATAMODE_HWFC - UART HWFC for data mode

   This option specifies requiring UART hardware flow cotrol for data mode or not.
   By default, UART hardware flow cotrol is required.

> 

Reviewer: typo "cotrol"

>    This option specifies requiring UART hardware flow cotroll for data mode or not.
>    By default, UART hardware flow cotroll is required.

Reviewer 2: ```suggestion
   This option specifies whether UART hardware flow control is required for data mode.
   By default, UART hardware flow control is required.
```

.. option:: CONFIG_SLM_DATAMODE_TERMINATOR - Pattern string to terminate data mode

   This option specifies a pattern string to terminate data mode.

>    Default pattern string is "+++".

Reviewer 2: ```suggestion
   The default pattern string is "+++".
```

.. option:: CONFIG_SLM_DATAMODE_SILENCE - Silence time to exit data mode

>    This option specifies the time (in seconds) of UART silence before and after the pattern
>    string used to exit data mode. Default value is 1s.

Reviewer 2: ```suggestion
   This option specifies the time (in seconds) of UART silence before and after the pattern string that is used to exit data mode. 
   The default value is 1 second.
```
